File: EcoMind/keyword_extraction.py
from rake_nltk import Rake

# Initialize the RAKE object
rake_nltk_var = Rake()

def extract_keywords(query):

    # Extract keywords by passing the text
    rake_nltk_var.extract_keywords_from_text(query)

    # Get the ranked phrases with their scores
    keywords_extracted = rake_nltk_var.get_ranked_phrases_with_scores()

    # Print the keywords and their scores
    for score, keyword in keywords_extracted:
        print(f"{keyword}: {score}")

    return keywords_extracted

File: EcoMind/keyword_extraction.js
const rake = require('rake-js');

// stopwords are the words in the query that will be ignored

function extractKeywords(query, stopwords) {
    // Extract keywords
    const keywordsExtracted = rake(query);

    // Print the keywords and their scores
    keywordsExtracted.forEach(keyword => {
        console.log(keyword);
    });

    return keywordsExtracted;
}

// Example usage
const query = `
Natural Language Processing (NLP) is a field of artificial intelligence that focuses on the interaction between computers and humans through natural language.
The ultimate objective of NLP is to read, decipher, understand, and make sense of the human languages in a manner that is valuable.
Most NLP techniques rely on machine learning to derive meaning from human languages.
`;

const extractedKeywords = extractKeywords(query, []);
console.log(extractedKeywords);